apps/openmw/mwworld/weather.cpp

```diff
@@ -427,29 +429,33 @@ void WeatherManager::update(float duration, bool paused)
     else
         mRendering->getSkyManager()->sunEnable();
 
-    // sun angle
-    float height;
-
-    //Day duration
-    float dayDuration = (mNightStart - 1) - mSunriseTime;
+    // Update the sun direction.  Run it east to west at 15 degrees south from overhead.
+    // The sun's speed at day and night will differ, since mSunriseTime and mNightStart
+    // mark when the sun is level with the horizon.
+    {
+        assert( mNightStart > mSunriseTime );
```

Comment:
I put the assert in to highlight that the code isn't trying to handle weird edge cases (like midnight being during the day, or sunrise/sunset being set to the same times).  If these assumptions are wrong, it's probably a soft rendering error, mapping the sun to the wrong spot.  If a throw is used, then the visual bug becomes a fatal error, and that seemed excessive.  Shouldn't we let the users play, if safe to do so?

There were a few parts I didn't dig into as much, since they had no visual effect:
Why is a visible sun trajectory calculated for nighttime, but hidden by the 'is_night' flag?  Why wasn't the sun just turned off like the moons are?  I think there were some comments relevant to this in the renderer sky.cpp file, but I didn't study that part of the system well enough to propose any simplifications.  The sun's direction could have been set to take the sun under the world plane instead, but since it seems to be turned off at night, I didn't change that part of the code.